STMReco/src/PlotSTMWaveformDigis_module.cc

```diff
@@ -72,12 +78,33 @@ namespace mu2e {
       std::cout << _channel.name() << " Pedestal = " << pedestal << std::endl;
     }
 
+    double x_min = 0;
+    double x_max = 0;
+    int n_bins = 0;
     for (const auto& waveform : *waveformsHandle) {
       histname.str("");
       histname << "evt" << event.event() << "_waveform" << count;
       histtitle.str("");
       histtitle << "Event " << event.event() << " Waveform " << count << " (" << _channel.name() << ")";
-      TH1F* _hWaveform = tfs->make<TH1F>(histname.str().c_str(), histtitle.str().c_str(), waveform.adcs().size(),0,waveform.adcs().size());
+
```

Comment:
Thanks, Rob. I've done that in commit [b8eeea8](https://github.com/Mu2e/Offline/pull/917/commits/b8eeea8644ceaf399e0a943dbb9b14d942a0e9e6). Is something like that what you had in mind?

Reply:
Yes.  That's what I had in mind.   In the branch for "waveform_time" is there a typo: xmin should be tmin?

And a followup question.  Is it safe to use just the event number as the histogram identifier?  Will you ever run this code on a set of files  with multiple runs/subruns so that event numbers might be duplicated?

Reply:
No that's not a typo. I want to define "waveform_time" as starting at 0. I willchange the variable names to make it clearer.

I have mixed feelings about adding the run and subrun number to the histogram name. I don't think we would run this code over multiple runs at once - it is supposed to be just a debugging tool - and it is easier to run a plotting macro on different files if we don't always need to specify a run and subrun number. However, we may want to compare waveforms between runs... I think I'd like to leave it as-is for now and revisit if we find we've made the wrong choice

Reply:
You might add a comment to notify future STM maintainers about that time convention.  If it's written up elsewhere, just reference it.

About the histogram identifier.  Your call.   Either way I wanted it to be a considered decision.

Reply:
... if you can tell the story well  enough with wisely chosen variable names, that better than a comment.

Reply:
I've chosen (hopefully) better variable names in [c1f1e6d](https://github.com/Mu2e/Offline/pull/917/commits/c1f1e6d35fb3b1442d1f8c63b6e92434ad5ed1e2)